### PR TITLE
[bitnami/prometheus] Fixes #2264

### LINKED ISSUE #2264: [bitnami/prometheus-operator] Containers section not rendered properly
**Which chart**:
bitnami/prometheus-operator version 0.14.10

**Describe the bug**
When adding a container in prometheus.containers section, an error is generated:
```
✗ helm -n mon template bitnami/prometheus-operator --version 0.14.0 -f prometheus.yaml --debug
install.go:158: [debug] Original chart version: "0.14.0"
install.go:175: [debug] CHART PATH: /Users/ajardan/Library/Caches/helm/repository/prometheus-operator-0.14.0.tgz

manifest_sorter.go:192: info: skipping unknown hook: "crd-install"
manifest_sorter.go:192: info: skipping unknown hook: "crd-install"
manifest_sorter.go:192: info: skipping unknown hook: "crd-install"
manifest_sorter.go:192: info: skipping unknown hook: "crd-install"
manifest_sorter.go:192: info: skipping unknown hook: "crd-install"
manifest_sorter.go:192: info: skipping unknown hook: "crd-install"
Error: YAML parse error on prometheus-operator/templates/prometheus/prometheus.yaml: error converting YAML to JSON: yaml: line 77: mapping values are not allowed in this context
helm.go:75: [debug] error converting YAML to JSON: yaml: line 77: mapping values are not allowed in this context
YAML parse error on prometheus-operator/templates/prometheus/prometheus.yaml
helm.sh/helm/v3/pkg/releaseutil.(*manifestFile).sort
	/private/tmp/helm-20200317-72180-1vcgqsd/src/helm.sh/helm/pkg/releaseutil/manifest_sorter.go:146
helm.sh/helm/v3/pkg/releaseutil.SortManifests
	/private/tmp/helm-20200317-72180-1vcgqsd/src/helm.sh/helm/pkg/releaseutil/manifest_sorter.go:106
helm.sh/helm/v3/pkg/action.(*Configuration).renderResources
	/private/tmp/helm-20200317-72180-1vcgqsd/src/helm.sh/helm/pkg/action/install.go:489
helm.sh/helm/v3/pkg/action.(*Install).Run
	/private/tmp/helm-20200317-72180-1vcgqsd/src/helm.sh/helm/pkg/action/install.go:230
main.runInstall
	/private/tmp/helm-20200317-72180-1vcgqsd/src/helm.sh/helm/cmd/helm/install.go:223
main.newTemplateCmd.func1
	/private/tmp/helm-20200317-72180-1vcgqsd/src/helm.sh/helm/cmd/helm/template.go:66
github.com/spf13/cobra.(*Command).execute
	/private/tmp/helm-20200317-72180-1vcgqsd/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826
github.com/spf13/cobra.(*Command).ExecuteC
	/private/tmp/helm-20200317-72180-1vcgqsd/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914
github.com/spf13/cobra.(*Command).Execute
	/private/tmp/helm-20200317-72180-1vcgqsd/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main
	/private/tmp/helm-20200317-72180-1vcgqsd/src/helm.sh/helm/cmd/helm/helm.go:74
runtime.main
	/usr/local/Cellar/go@1.13/1.13.8/libexec/src/runtime/proc.go:203
runtime.goexit
	/usr/local/Cellar/go@1.13/1.13.8/libexec/src/runtime/asm_amd64.s:1357
```

**To Reproduce**
1. Add this to prometheus.containers section in prometheus.yaml:
```
prometheus:
  containers:
    - name: blah
```
2. Run:
```
$ helm template bitnami/prometheus-operator --version 0.14.0 -f prometheus.yaml --debug
```
3. See the error
**Expected behavior**
I would expect to see the generated templates this char is going to deploy, including the newly added container

**Version of Helm and Kubernetes**:
Helm 3.1.2
Kubernetes 1.18.0
- Output of `helm version`:

```
version.BuildInfo{Version:"v3.1.2", GitCommit:"d878d4d45863e42fd5cff6743294a11d28a9abce", GitTreeState:"clean", GoVersion:"go1.13.8"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-26T06:16:15Z", GoVersion:"go1.14", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.11-gke.3", GitCommit:"71631703d37fb1b4b333abd427deae839aeb2032", GitTreeState:"clean", BuildDate:"2020-03-24T18:54:48Z", GoVersion:"go1.12.17b4", Compiler:"gc", Platform:"linux/amd64"}
```

### PR DESCRIPTION
**Pull request by ni-ajardan**
**Description of the change**

One can add extra containers as sidecars to Prometheus

**Benefits**

This will allow people to customise existing containers, or add more sidecars for other functionalities (Istio, etc)

**Possible drawbacks**

Hmm, none I can think of.

**Applicable issues**

  - fixes #2264 

**Additional information**

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [ ] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [ ] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/prometheus-operator/templates/prometheus/prometheus.yaml [hunk 1]
```diff
@@ -177,7 +177,7 @@ spec:
           protocol: TCP
     {{- end }}
     {{- if .Values.prometheus.containers }}
-    {{- include "prometheus-operator.tplValue" (dict "value" .Values.prometheus.containers "context" $) | indent 4 }}
+    {{ toYaml .Values.prometheus.containers | indent 4 }}
```
Comment (juan131) [inline bitnami/prometheus-operator/templates/prometheus/prometheus.yaml:180]: The `"prometheus-operator.tplValue"` macro should detect that `.Values.prometheus.containers` is a YAML and render it accordingly. See:
```
{{/*
Renders a value that contains template.
Usage:
{{ include "prometheus-operator.tplValue" ( dict "value" .Values.path.to.the.Value "context" $) }}
*/}}
{{- define "prometheus-operator.tplValue" -}}
    {{- if typeIs "string" .value }}
        {{- tpl .value .context }}
    {{- else }}
        {{- tpl (.value | toYaml) .context }}
    {{- end }}
{{- end -}}
```
Replacing `indent` with `nindent` should be enough:
```diff
-     {{- include "prometheus-operator.tplValue" (dict "value" .Values.prometheus.containers "context" $) | indent 4 }}
+     {{- include "prometheus-operator.tplValue" (dict "value" .Values.prometheus.containers "context" $) | nindent 4 }}
```

**Comment by juan131** [comment]
@ni-ajardan please also rebase your branch from master and solve the conflicts

**Comment by ni-ajardan** [comment]
@juan131 Thanks a lot for your quick response and advice ! Does this look better ?

**Comment by juan131** [comment]
Waiting for CI/CD to run tests and update image tags

**Comment by ni-ajardan** [comment]
Thanks a lot @juan131 !

**Pull request merged**